### PR TITLE
Allow for exact match to lookup file id and confirm that both filenam…

### PR DESCRIPTION
**Pull request by johnathan-rodriguez**
…e and folder name match the inputted value

### CODE REVIEW

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 1]
```diff
@@ -43,24 +43,12 @@ def get_args():
     return parser.parse_args()
 
 
-def set_environment_variables(args):
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:46]: is not necessary. The application already passes in BOX_APPLICATION_CREDENTIALS as args.service account

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 2]
```diff
@@ -43,24 +43,12 @@ def get_args():
...
-
-
 def extract_file_name_from_source_full_path(source_full_path):
     """
     Use the file name provided in the source_file_name variable. Should be run only
     if a destination_file_name is not provided.
     """
-    destination_file_name = os.path.basename(source_full_path)
-    return destination_file_name
+    return os.path.basename(source_full_path)
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:51]: just cleaned up a bit- no need for the return value to be stored as a variable if it is just going to be immediately returned

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 3]
```diff
@@ -132,13 +120,12 @@ def determine_destination_name(
         source_full_path=source_full_path,
         file_number=file_number,
     )
-    destination_name = combine_folder_and_file_name(
+    return combine_folder_and_file_name(
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:123]: Same cleanup concept here

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 4]
```diff
@@ -132,13 +120,12 @@ def determine_destination_name(
...
         destination_folder_name, destination_file_name
     )
-    return destination_name
 
 
-def find_box_file_names(client, source_folder_name, source_file_name):
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:141]: `source_file_name` was not being reference in the function

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 5]
```diff
@@ -207,41 +194,53 @@ def get_client(service_account):
         client.user().get()
         return client
     except BoxOAuthException as e:
-        print(
-            f"Error accessing Box account with pervice account "
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:211]: there was a typo and non of the values in 
            f"Error accessing Box account with pervice account "
            f"developer_token={developer_token}; client_id={client_id}; "
            f"client_secret={client_secret}"
were declared anywhere else

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 6]
```diff
@@ -207,41 +194,53 @@ def get_client(service_account):
...
-            f"developer_token={developer_token}; client_id={client_id}; "
-            f"client_secret={client_secret}"
+        print("Error accessing Box account with service account ")
+        raise e
+
+
+def get_folder(client, folder_name):
+    """
+    Returns the folder id for the Box client if it exists.
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:203]: Added a function because the sdk query will return anything that matches a string with the query params. This ensures parent/subfolder actually matches the `subfolder` name. Box does not return the full paths of folders.

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 7]
```diff
@@ -207,41 +194,53 @@ def get_client(service_account):
...
+        if not folder_id:
+            raise FileNotFoundError(f"Folder {source_folder_name} not found")
+        files = client.folder(folder_id).get_items()
         for _file in files:
-            return (_file.name, _file.id)
+            if _file.name == source_file_name:
+                print(f"Found file {source_file_name} with ID {_file.id}")
+                return _file.name, _file.id
+
+        raise FileNotFoundError(f"File {source_file_name} not found in folder {source_folder_name}")
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:237]: Part of the issue that customers were seeing was than if the folder was found but no files existed within the folder it returned a difficult to read error. I included logging to assist with the troubleshooting process and added logs in plain english

File: shipyard_blueprints/box/shipyard_box/cli/download.py [hunk 8]
```diff
@@ -253,13 +252,12 @@ def main():
...
         os.makedirs(destination_folder_name)
 
-    client = get_client(service_account=service_account)
+    client = get_client(service_account=args.service_account)
 
     if source_file_name_match_type == "regex_match":
         file_objs = find_box_file_names(
             client=client,
             source_folder_name=source_folder_name,
-            source_file_name=source_file_name,
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/box/shipyard_box/cli/download.py:262]: was not being referenced in the function

**Review by blakeburch — APPROVED** [review]
